results_spreadsheet.py

```diff
@@ -0,0 +1,136 @@
+output_excel = "./analysis/results_spreadsheet.xlsx"
+mapping_file = "./data/county2zone.csv"
+
+# Technologies to collapse
+agg_techs = ["wind-ons", "wind-ofs", "pv", "csp", "hyd", "egs", "coal", "gas"]
+
+
+# === Helpers ===
+def load_region_map(path):
+    df = pd.read_csv(path)
+    if "ba" in df.columns and "r" not in df.columns:
+        df = df.rename(columns={"ba": "r"})
+    return df.set_index("r")["state"].to_dict()
+
+
+def clean_and_aggregate(df, scenario, region_to_state):
+    # Normalize headers
+    df.columns = df.columns.str.strip().str.lower()
+
+    # Map either 'r' or 'ba' to state (whichever exists)
+    col = {"r", "ba"}.intersection(df.columns)
+    if col:
+        col = col.pop()
+        df["state"] = df[col].map(region_to_state)
+        df = df.drop(columns=[col])
+
+    # collapse i
+    if "i" in df.columns:
+        df["i"] = df["i"].astype(str).str.strip().str.lower()
+        for tech in agg_techs:
+            if tech in ["coal", "gas"]:
+                df.loc[
+                    df["i"].str.contains(tech, case=False, na=False)
+                    & ~df["i"].str.contains("ccs", case=False, na=False),
+                    "i",
+                ] = tech
+                df.loc[
+                    df["i"].str.contains(tech, case=False, na=False)
+                    & df["i"].str.contains("ccs", case=False, na=False),
+                    "i",
+                ] = f"{tech}_ccs"
+            else:
+                df.loc[df["i"].str.contains(tech, case=False, na=False), "i"] = tech
+
+    # add Scenario
+    df.insert(0, "scenario", scenario)
+
+    # aggregate value
+    # Identify numeric columns to aggregate
+    if "value" in df.columns:
+        agg_cols = ["value"]
+    elif {"tons", "md", "damage_$", "mortality"}.issubset(df.columns):
+        agg_cols = ["tons", "md", "damage_$", "mortality"]
+    else:
+        agg_cols = []
+
+    if agg_cols:
+        # Group by all non-aggregated columns
+        group_cols = [c for c in df.columns if c not in agg_cols]
+        df = df.groupby(group_cols, as_index=False)[agg_cols].sum()
+    else:
+        df = df.drop_duplicates()
+
+    return df
+
+
+# === Main ===
+summary_df = pd.read_csv(summary_file)
+region_to_state = load_region_map(mapping_file)
+csv_types = [c for c in summary_df.columns if c.lower() != "scenario"]
+
+tabs = {csv_name: [] for csv_name in csv_types}
+
+# Wrap scenario loop in tqdm progress bar
+for _, row in tqdm(
+    summary_df.iterrows(), total=len(summary_df), desc="Processing scenarios"
+):
+    scenario = row["Scenario"]
+    for csv_name in csv_types:
+        csv_path = os.path.join(base_path, scenario, csv_name)
+        if not os.path.exists(csv_path):
+            continue
+        try:
+            df = pd.read_csv(csv_path)
+            df = clean_and_aggregate(df, scenario, region_to_state)
+            tabs[csv_name].append(df)
+        except Exception as e:
+            print(f"Error processing {csv_path}: {e}")
+
```

Comment:
Several thoughts:
1. Generally, it's best practice to specify exactly _which_ errors are being handled because it can cause unintended errors to sneak past you.  If you expect this to happen for a file that doesn't exist or can't be found, consider a `FileNotFound` error (you can catch multiple errors at once).
2. You can also consolidate your code and use the try/except block handle the problem solved in the IF statement.
3. `os.path.join` is completely workable, but a little inelegant. I prefer `pathlib` for this task.

```suggestion
    for csv_name in csv_types:
        csv_path = os.path.join(base_path, scenario, csv_name)
        try:
            df = pd.read_csv(csv_path)
            df = clean_and_aggregate(df, scenario, region_to_state)
            tabs[csv_name].append(df)
        except FileNotFoundError as e:
            print(f"Error processing {csv_path}: {e}")
```